contrib/drivers/ydb/ydb.go

```diff
@@ -0,0 +1,148 @@
+package ydb
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/gogf/gf/v2/database/gdb"
+	"github.com/gogf/gf/v2/errors/gcode"
+	"github.com/gogf/gf/v2/errors/gerror"
+	"github.com/gogf/gf/v2/text/gregex"
+	"github.com/gogf/gf/v2/util/gutil"
+	_ "github.com/ydb-platform/ydb-go-sdk/v3"
+)
+
+// Driver is the driver for ydb database.
+type Driver struct {
+	*gdb.Core
+}
+
+func init() {
+	if err := gdb.Register(`ydb`, New()); err != nil {
+		panic(err)
+	}
+}
+
+// New create and returns a driver that implements gdb.Driver, which supports operations for YDB.
+func New() gdb.Driver {
+	return &Driver{}
+}
+
+// New creates and returns a database object for ydb.
+// It implements the interface of gdb.Driver for extra database driver installation.
+func (d *Driver) New(core *gdb.Core, node *gdb.ConfigNode) (gdb.DB, error) {
+	return &Driver{
+		Core: core,
+	}, nil
+}
+
+// Open creates and returns an underlying sql.DB object for ydb.
+// Note that it converts time.Time argument to local timezone in default.
+func (d *Driver) Open(config *gdb.ConfigNode) (db *sql.DB, err error) {
+	var (
+		source               string
+		underlyingDriverName = "ydb"
+	)
+	// [username[:password]@][protocol[(address)]]/dbname[?param1=value1&...&paramN=valueN]
+	if config.Link != "" {
+		// ============================================================================
+		// Deprecated from v2.2.0.
+		// ============================================================================
+		source = config.Link
+		// Custom changing the schema in runtime.
+		if config.Name != "" {
+			source, _ = gregex.ReplaceString(`/([\w\.\-]+)+`, "/"+config.Name, source)
+		}
+	} else {
+		source = fmt.Sprintf(
+			"%s:%s@%s(%s:%s)/%s?charset=%s",
+			config.User, config.Pass, config.Protocol, config.Host, config.Port, config.Name, config.Charset,
+		)
+		if config.Timezone != "" {
+			if strings.Contains(config.Timezone, "/") {
+				config.Timezone = url.QueryEscape(config.Timezone)
+			}
+			source = fmt.Sprintf("%s&loc=%s", source, config.Timezone)
```

Comment:
Please make sure the `ydb` supports `local` parameters, which is supported by `mysql` driver.